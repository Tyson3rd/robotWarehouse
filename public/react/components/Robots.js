import React, { useState } from 'react';

const Robots = (props) => {



    return (
       <div>
           <h2>All Robots:</h2>
            <table class="table bg-white rounded shadow-sm  table-hover">
            <thead>
                
                                        <tr>
                                            <th scope="col">#</th>
                                            <th scope="col">Robot Name</th>
                                            <th scope="col">Type</th>
                                            
                                        </tr>
                                    </thead>
			{
			
        props.robots.map((robot, i) => {
        return (<tbody>
            <tr key={i}>
                <th scope="row">{robot.id}</th>
                <td>{robot.name}</td>
                <td>{robot.type}</td>
    
            </tr>
        </tbody>)
        })
            }
		</table>
       </div> 
    )
}
module.exports = Robots