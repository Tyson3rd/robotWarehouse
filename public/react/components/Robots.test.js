import React from 'react';
import renderer from 'react-test-renderer';

import  Robots  from './Robots';

describe("Robots component", () => {

  

  test("renders Robots component as expected", () => {
    const robots = ['R2D2', 'T-800'];


    const component = renderer.create(<Robots robots={robots} />)

    const snapshot = component.toJSON();

    console.log("What does our snapshot look like?  ", snapshot)

    expect(snapshot).toMatchSnapshot();
  })

  

  test("Contains items passed down as props", () => {
    const robots = ['R2D2', 'T-800'];
    
    const component = renderer.create(<Robots robots={robots}/>);
    const treeEl = component.toTree()

    

    expect(treeEl.props.robots.length).toBe(2)
  })

})