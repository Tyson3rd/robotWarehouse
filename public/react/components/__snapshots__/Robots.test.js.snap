// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Robots component renders Robots component as expected 1`] = `
<div>
  <h2>
    All Robots:
  </h2>
  <table
    class="table bg-white rounded shadow-sm  table-hover"
  >
    <thead>
      <tr>
        <th
          scope="col"
          width="50"
        >
          #
        </th>
        <th
          scope="col"
        >
          Robot Name
        </th>
        <th
          scope="col"
        >
          Type
        </th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <th
          scope="row"
        />
        <td />
        <td />
      </tr>
    </tbody>
    <tbody>
      <tr>
        <th
          scope="row"
        />
        <td />
        <td />
      </tr>
    </tbody>
  </table>
</div>
`;
